tools/src/test/java/org/apache/kafka/tools/consumer/ConsoleConsumerTest.java

```diff
@@ -239,4 +296,130 @@ public void shouldWorkWithoutTopicOption() throws IOException {
         verify(mockConsumer).subscribe(any(Pattern.class));
         consumer.cleanup();
     }
+
+    @ClusterTest(types = {Type.KRAFT, Type.CO_KRAFT}, brokers = 3)
+    public void testTransactionLogMessageFormatter() throws Exception {
+        try (Admin admin = cluster.createAdminClient()) {
+
+            List<TransactionLogTestcase> testcases = generateTestcases();
```

Comment:
This test does NOT include the scenario we do care.

1. we should verify the topic `__transaction_state` has data
2. we should verify the new formatter can parse the data of `__transaction_state`
3. we can use arbitrary data in producing transaction since all we want to check is the data in `__transaction_state`